libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -48,11 +58,16 @@ public void close() throws IOException {
         ((Closeable) tracingTelemetry).close();
     }
 
-    // Visible for testing
-    Span getCurrentSpan() {
+    private Span getCurrentSpanInternal() {
         return tracerContextStorage.get(TracerContextStorage.CURRENT_SPAN);
     }
 
+    public SpanContext getCurrentSpan() {
+        return tracerContextStorage.get(TracerContextStorage.CURRENT_SPAN) == null
```

Comment:
Super nit (no need to ask context storage twice):

```suggestion
        final Span span = tracerContextStorage.get(TracerContextStorage.CURRENT_SPAN);
        return (span == null) ? null :  new SpanContext(span);
```